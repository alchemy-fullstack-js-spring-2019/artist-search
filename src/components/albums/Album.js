import React from 'react';
import PropTypes from 'prop-types';

function Album({ album, image  }) {
  return (
    <>
    <p>{album}</p>
    {image && <img style={{ width: '100px', height: 'auto' }} src={image} ></img>}
    </>
  );

}

Album.propTypes = {
  album: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Album;
