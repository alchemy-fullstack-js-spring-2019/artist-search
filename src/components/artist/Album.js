import React from 'react';
import PropTypes from 'prop-types';
import Art from './Art';
import AlbumInfo from './AlbumInfo';

function Album({ album }) {
  return (
    <>
      <Art id={album.id} archive={album['cover-art-archive'].front} />
      <AlbumInfo title={album.title} year={album.date} />
    </>
  );
}

Album.propTypes = {
  album: PropTypes.object.isRequired
};

export default Album;
