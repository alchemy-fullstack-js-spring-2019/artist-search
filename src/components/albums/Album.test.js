import React from 'react';
import { shallow } from 'enzyme';
import Album from './Album';

describe('Album component',  ()=>{
  it('renders stuff', ()=>{
    const wrapper = shallow(<Album album='test album' image='test image'/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
