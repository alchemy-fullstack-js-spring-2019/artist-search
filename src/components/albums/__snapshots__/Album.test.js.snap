// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album component renders stuff 1`] = `
<Fragment>
  <p>
    test album
  </p>
  <img
    src="test image"
    style={
      Object {
        "height": "auto",
        "width": "100px",
      }
    }
  />
</Fragment>
`;
